turbopy/core.py

```diff
@@ -651,20 +652,29 @@ def set_value_from_keys(self, var_name, options):
         raise (KeyError("Grid configuration for " + var_name
                         + " not found."))
 
-    def generate_field(self, num_components=1):
+    def generate_field(self, num_components=1, placement_of_points="edge-centered"):
         """Returns squeezed :class:`numpy.ndarray` of zeros with
         dimensions :class:`Grid.num_points` and `num_components`.
 
         Parameters
         ----------
         num_components : int, defaults to 1
             Number of vector components at each point.
+        placement_of_points : str, defaults to "edge-centered"
+            Designate position of points on grid
         Returns
         -------
         :class:`numpy.ndarray`
             Squeezed array of zeros.
         """
-        return np.squeeze(np.zeros((self.num_points, num_components)))
+        number_of_field_points = None
+        if placement_of_points == "edge-centered":
+            number_of_field_points = self.num_points
+        elif placement_of_points == "cell-centered":
+            number_of_field_points = self.num_points - 1
+        else:
+            raise ValueError("Unknown placement option specified")
+        return np.squeeze(np.zeros((number_of_field_points, num_components)))
```

Comment:
Same for this... looks like you are trying to merge changes that are already in `main`.